src/Http/Http.Extensions/src/ProblemDetailsJsonContext.cs

```diff
@@ -9,6 +9,8 @@ namespace Microsoft.AspNetCore.Http;
 
 [JsonSerializable(typeof(ProblemDetails))]
 [JsonSerializable(typeof(HttpValidationProblemDetails))]
+// ExtensionData
+[JsonSerializable(typeof(IDictionary<string, object?>))]
```

Comment:
I can potentially avoid adding by using the [JsonMetadataServices.ObjectConvert](https://source.dot.net/#System.Text.Json/System/Text/Json/Serialization/Metadata/JsonMetadataServices.Converters.cs,157) in our [ProblemDetailsJsonConverter](https://github.com/dotnet/aspnetcore/blob/afd597e7fbfbe3440dda8884d355af88a8f33a1e/src/Shared/ProblemDetails/ProblemDetailsJsonConverter.cs#L84), however, the documentation explicitly states that this api should not be called directly.

Reply:
I don't think there's a reason to treat extension data differently. I'd leave as-is. If necessary we could revisit this pattern in a broader discussion about whether to include any types implicitly (e.g. primitive types like string, int, etc). IMO we shouldn't.

There's no practical harm in using that converter, but it's better to follow the docs and not call `JsonMetadataServices` APIs directly.

Reply:
Thanks.

> I'd leave as-is

That means adding the `[JsonSerializable(typeof(IDictionary<string, object?>))]`, right?